students/BrianE/session03/list_lab.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/env python3
+
+"""
+    list_lab.py
+"""
+
+
+def display_fruits(fruits):
+    """
+    Print fruit list contents to terminal
+    :param fruits: list of fruits
+    :return: None
+    """
+
+    print("Current fruit: {}".format(fruits))
+
+
+def delete_fruits(fruits):
+    """
+    Delete fruit item from list
+    :param fruits: string
+    :return: None
+    """
+
+    print(fruits)
+    fruit_to_delete = input("Enter a fruit to delete: \n\n >>>")
+    try:
+        index = fruits.index(fruit_to_delete)
+        del fruits[index]
+    except ValueError as e:
+        print(e)
+
+
+def cleanup_list(fruits):
+    """
+    Remove fruits from list if user does not like
+    :param fruits: list of fruits
+    :return: None
+    """
+
+    for fruit in fruits:
```

Comment:
You will want to be careful here. Everytime we say no and remove a fruit, we jump up an extra one (because we are not accounting for the fruit we just removed). Try this out and see how it acts. 

Hint: Try to iterate through the fruits list where you can control the iterator, otherwise it will jump an extra spot when you don't want it to.

Reply:
Corrected, but I don't think this is the best solution. I attempted to subtract 1 from the value of i when the user answers "no". The next item in the list was still skipped and I cannot figure out why.

Instead, I've created a new list of items to delete and I iterate through them afterwards, removing from the original list. Let me know what you think.

Reply:
My apologies, you would not be able to modify the iterator value itself, you would likely need to keep another variable to track it. Your way of solving this works as well, just keep the note about memory in mind!